pkg/controller/infrastructure/infraflow/utils_test.go

```diff
@@ -0,0 +1,116 @@
+package infraflow_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/ptr"
+
+	awsclient "github.com/gardener/gardener-extension-provider-aws/pkg/aws/client"
+	. "github.com/gardener/gardener-extension-provider-aws/pkg/controller/infrastructure/infraflow"
+)
+
+var _ = Describe("FindExisting", func() {
```

Comment:
To fully test FindExisting() shouldn't we also add a test with a selector?

Reply:
Addressed in latest commit